### ISSUE TITLE
decimal.Decimal is parsed as string

### ISSUE THREAD

**Issue by jzyinq** [glothriel/grf#24]
Given our example product model:
```go
type Product struct {
	models.BaseModel

	Name        string          `json:"name" gorm:"size:191;column:name"`
	Description string          `json:"description" gorm:"type:text;column:description"`
	Price       decimal.Decimal `json:"price" gorm:"type:decimal(19,4)"`
}
```

`decimal.Decimal` is not handled properly:
- post expects `string` instead of `float` in json payload
```json
{
	"errors": {
		"price": [
			"Field `price` is not a string"
		]
	}
}
```
- there is no precision setting that would validate decimal 
```json
[
	{
		"id": "cd3116b1-1bfd-431d-80a0-ac11232855d2",
		"name": "TestProduct",
		"price": "2.8"
	},
	{
		"id": "e54d89ba-206f-4945-9ef3-2c23b27eb837",
		"name": "TestProduct",
		"price": "2.801"
	},
	{
		"id": "2d99fee3-64fd-43e0-b2ed-eb064088ce9c",
		"name": "TestProduct",
		"price": "2.80005"
	}
]
```

**Comment by glothriel** [comment]
In general the Decimal type IS THERE because you don't want to use a float, as it's not suitable for currency calculations. Afaik the issue you are facing is related to how [SQLITE handles Decimal Type](https://stackoverflow.com/questions/21757722/how-to-use-sqlite-decimal-precision-notation)

TLDR;
> Of course, any value entered will be stored, even if it doesn't keep to these rules, which means that the field definition is basically documentation.

**Comment by glothriel** [comment]
So:
* String type on input and output is on purpose
* The lack of precision settings enforcing is due to how SQLite works, should work fine on other DB. In case of SQLite you have to handle this in your code, in our example:
```
func (p *Product) BeforeSave(tx *gorm.DB) error {
    // Check if the price has more than 2 decimal places
    if p.Price.Exponent() < -2 {
        return errors.New("price cannot have more than 2 decimal places")
    }
    return nil
}
```
